include/tilck/common/boot.h

```diff
@@ -0,0 +1,27 @@
+/*
+ * The following two defines are used for `Type`
+ * of EFI_MEMORY_DESCRIPTOR struct. See
+ * `EfiToMultibootMemType()` for more details.
+ *
+ * The next two defines (suffixed by "REQ") are
+ * used for the `Type`s which also additionally
+ * have `Attribute` as EFI_MEMORY_RUNTIME. This
+ * value for `Attribute` field marks the memory
+ * regions required at runtime and hence should
+ * provided with a virtual address for calling
+ * SetVirtualAddressMap(). See
+ * setup_uefi_runtime_services() for more
+ * details.
+ */
+
+#define TILCK_BOOT_EFI_RUNTIME_CODE MULTIBOOT_MEMORY_RESERVED+1
+#define TILCK_BOOT_EFI_RUNTIME_DATA MULTIBOOT_MEMORY_RESERVED+2
+
+#define TILCK_BOOT_EFI_RUNTIME_CODE_REQ MULTIBOOT_MEMORY_RESERVED+3
```

Comment:
Why do we need to have this CODE_REQ and DATA_REQ types? Couldn't we just map them to the EFI_RUNTIME_CODE and DATA types? I haven't check the spec myself. I mean, do we need to keep mapped ALL of those region types or just the REQ ones? If it's just the REQ ones, we should totally skip the first two. If we need them all, why it's so important to distinguish them? 

Btw, for the sake of the tests, until everything works, keep all of those mem types mapped. It will be easier to remove them later and check if that's still OK.

Reply:
You are correct, we can skip the first two types. I tried doing that in the latest diff and it works.